### PR TITLE
Improve euristic about editable packages

### PR DESCRIPTION
**Pull request by vhaldemar**
Fix a problem with classifying packages installed via conda as editables

### CODE REVIEW

File: pylzy/lzy/py_env/py_env_provider.py [hunk 1]
```diff
@@ -378,3 +376,51 @@ def _get_module_parents(module: ModuleType) -> Iterable[ModuleType]:
...
+    file_scheme = 'file://'
+
+    if not url.startswith(file_scheme):
+        return False
+
+    raw_path = url[len(file_scheme):]
+
+    # idk if there may be relative paths in direct_url.json,
+    # but if it is, next code will work badly with it.
+    if not raw_path.startswith('/'):
```
Comment (ArtoLord) [inline pylzy/lzy/py_env/py_env_provider.py:391]: This path in windows can contain '\' and other windows stuff.
We can use os.path here or just skip this check

File: pylzy/lzy/py_env/py_env_provider.py [hunk 2]
```diff
@@ -378,3 +376,51 @@ def _get_module_parents(module: ModuleType) -> Iterable[ModuleType]:
...
         parts.pop()
+
+
+def _check_url_is_local_file(url: str) -> bool:
+    file_scheme = 'file://'
+
+    if not url.startswith(file_scheme):
+        return False
+
+    raw_path = url[len(file_scheme):]
```
Comment (ArtoLord) [inline pylzy/lzy/py_env/py_env_provider.py:387]: Why not use `urlparse` here

File: pylzy/lzy/py_env/py_env_provider.py [hunk 3]
```diff
@@ -378,3 +376,51 @@ def _get_module_parents(module: ModuleType) -> Iterable[ModuleType]:
...
+    distribution = importlib_metadata.distribution(package_name)
+
+    direct_url_str = distribution.read_text('direct_url.json')
+    if not direct_url_str:
+        # there is not direct_url.json
+        return False
+
+    direct_url_data = json.loads(direct_url_str)
+
+    url = direct_url_data.get('url', '')
```
Comment (ArtoLord) [inline pylzy/lzy/py_env/py_env_provider.py:416]: Why get empty line here?
We can return `None`and return False early

File: pylzy/lzy/py_env/py_env_provider.py (hunk 2 above)
Comment (vhaldemar) [reply pylzy/lzy/py_env/py_env_provider.py:387]: I think it is kinda too excess to use this microscope here.

**Pull request merged**